src/ert/config/gen_data_config.py

```diff
@@ -15,36 +15,39 @@ class GenDataConfig(ResponseConfig):
 
     def __post_init__(self) -> None:
         self.report_steps = (
-            SortedList([0])
+            SortedList()
             if not self.report_steps
             else SortedList(set(self.report_steps))
         )
 
     def read_from_file(self, run_path: str, _: int) -> xr.Dataset:
-        errors = []
-        datasets = []
-        filename_fmt = self.input_file
-        _run_path = Path(run_path)
-        for report_step in self.report_steps:
-            filename = filename_fmt % report_step
-            if not Path.exists(_run_path / filename):
-                errors.append(f"{self.name} report step {report_step} missing")
-                continue
-
+        def _read_file(filename, report_step):
             data = np.loadtxt(_run_path / filename, ndmin=1)
             active_information_file = _run_path / (filename + "_active")
             if active_information_file.exists():
                 index_list = (np.loadtxt(active_information_file) == 0).nonzero()
                 data[index_list] = np.nan
-            datasets.append(
-                xr.Dataset(
+            return xr.Dataset(
                     {"values": (["report_step", "index"], [data])},
                     coords={
                         "index": np.arange(len(data)),
                         "report_step": [report_step],
                     },
                 )
-            )
+
+        errors = []
+        datasets = []
+        filename_fmt = self.input_file
+        _run_path = Path(run_path)
+        if self.report_steps:
+            for report_step in self.report_steps:
+                filename = filename_fmt % report_step
+                if not Path.exists(_run_path / filename):
+                    errors.append(f"{self.name} report step {report_step} missing")
+                    continue
+                datasets.append(_read_file(filename, report_step))
+        else:
+            datasets.append(_read_file(filename_fmt, 0))
```

Comment:
We are not guarding against the file not existing here, think we should provide a reasonable error message in this scenario.